spec/unit/application/client_spec.rb

```diff
@@ -320,9 +320,14 @@
         end
         expect(@pipe[0].gets).to eq("started\n")
         Process.kill("TERM", pid)
-        Process.wait
-        sleep 1 # Make sure we give the converging child process enough time to finish
-        expect(IO.select([@pipe[0]], nil, nil, 0)).not_to be_nil
+        Process.wait(pid)
+        output = nil; i = 0
+        while output.nil? && i < 10
+          sleep 1
+          i += 1
+          output = IO.select([@pipe[0]], nil, nil, 0)
```

Comment:
May want to rename the `output` variable since it's not actually the output, but is a list of readable IO objects. Maybe could call it `child_wrote_to_stdout` or something? You could potentially roll the timeout into the select and remove the loop, but if this is working now, I don't see a reason you need to change it.